akka-javasdk/src/main/java/akka/javasdk/agent/SessionHistoryMessage.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.agent;
+
+import akka.javasdk.agent.SessionMessage.ToolCallInteraction;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import java.util.List;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@type")
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = SessionHistoryMessage.UserMessage.class, name = "HUM"),
+  @JsonSubTypes.Type(value = SessionHistoryMessage.AiMessage.class, name = "HAIM")})
+public sealed interface SessionHistoryMessage {
```

Comment:
Open for discussion on how to name it, but we need a separate type with `timestamp` field.

Reply:
I'm confused. We have pretty much the same thing in SessionMessage. Can't we add the timestamp to the SessionMessage? It might be even more correct to let the agent create the timestamp than the SessionMemoryEntity?

Reply:
Ohh, I thought that we should have a timestamp from the entity, ok that would simplify a lot